src/test/java/io/vlingo/lattice/exchange/TestExchangeReceiver2.java

```diff
@@ -21,10 +21,20 @@ public TestExchangeReceiver2(final TestUntil until, final ConcurrentLinkedQueue<
     this.results = results;
   }
 
+  public TestExchangeReceiver2(final ConcurrentLinkedQueue<Object> results) {
+	this.until = null;
+	this.results = results;
+  }
+
   @Override
   public void receive(final LocalType2 message) {
     System.out.println("TestExchangeReceiver2 receiving: " + message);
     results.add(message);
+
+    if (until == null) {
```

Comment:
This should be an `AccessSafely`. For example, do this inside `AccessSafely`: `results.add(message);`